pkg/plugins/runtime/gateway/generator.go

```diff
@@ -256,6 +256,10 @@ func (g Generator) generateLDS(ctx xds_context.Context, info GatewayListenerInfo
 	for _, hostInfo := range hostInfos {
 		gatewayHosts = append(gatewayHosts, hostInfo.Host)
 	}
+	if info.Listener.CrossMesh {
+		//for cross-mesh, we can only add one listener filter chain as there will not be any (usable) SNI available for filter chain matching
```

Comment:
`gocritic` is complaining

```suggestion
		// for cross-mesh, we can only add one listener filter chain as there will not be any (usable) SNI available for filter chain matching
```